examples/js/examples.js

```diff
@@ -300,4 +314,96 @@ function indentLines (input, indentation) {
   return lines.join('\n');
 }
 
+/**
+ * Creates and adds an "Open in CodePen" button
+ *
+ * @param {String} exampleIndex - the example number, if there are multiple examples
+ * @param {String} exampleHeaderId - the example header to place the button next to
+ * @param {String} exampleCodeId - the example html code
+ * @param {String} exampleFilesId - the element containing all relevent CSS and JS file
+ */
+function addOpenInCodePenForm (exampleIndex, exampleHeaderId, exampleCodeId, exampleFilesId) {
+  var jsonInputId = 'codepen-data-ex-' + exampleIndex;
+  var buttonId = exampleCodeId + '-codepenbutton'
+
+  var form = document.createElement('form');
+  form.setAttribute('action', 'https://codepen.io/pen/define');
+  form.setAttribute('method', 'POST');
+  form.setAttribute('target', '_blank');
+
+  var input = document.createElement('input');
+  input.setAttribute('id', jsonInputId);
+  input.setAttribute('type', 'hidden');
+  input.setAttribute('name', 'data');
+
+  var button = document.createElement('button');
+  button.innerText = 'Open In CodePen';
+
+  form.appendChild(input);
+  form.appendChild(button);
+
+  var exampleHeader = document.getElementById(exampleHeaderId);
+  exampleHeader.parentNode.insertBefore(form, exampleHeader.nextSibling);
+
+  // Correct the indentation for the example html
+  var indentedExampleHtml = document.getElementById(exampleCodeId).innerHTML;
+  indentedExampleHtml = indentedExampleHtml.replace(/^\n+/, '');
+  var indentation = indentedExampleHtml.match(/^\s+/)[0];
+  var exampleHtml = indentedExampleHtml.replace(new RegExp('^' + indentation, 'gm'), '');
+
+  var postJson = {
+    html: exampleHtml,
+    css: '',
+    js: '',
+    head: '<base href="' + location.href + '">'
+  };
+
+  var totalFetchedFiles = 0;
+  var fileLinks = document.querySelectorAll('#' + exampleFilesId + ' a');
+
+  for (let fileLink of fileLinks) {
+
+      var request = new XMLHttpRequest();
+
+      request.open('GET', fileLink.href, true);
+      request.onload = function() {
+          var href = this.responseURL;
+          if (this.status >= 200 && this.status < 400) {
+              if (href.indexOf('css') !== -1) {
+                postJson.css = postJson.css.concat(this.response);
+              }
+              if (href.indexOf('js') !== -1) {
+                postJson.js = postJson.js.concat(this.response);
+              }
+              totalFetchedFiles++;
+          }
+          else {
+              hideButton(buttonId, "Could not load resource: " + href);
+          }
+      };
+      request.onerror = function() {
+          hideButton(buttonId, "Could not load resource: " + fileLink.href);
+      };
+      request.send();
+  }
+
+    var timerId = setInterval(() => {
+        console.log(totalFetchedFiles);
+        if (totalFetchedFiles === fileLinks.length) {
+            document.getElementById(jsonInputId).value = JSON.stringify(postJson);
+            clearInterval(timerId);
+        }
+    }, 500);
```

Comment:
Not a thing that needs to be fixed, but it looks like async/await is supported in all modern browsers. If we wanted to avoid using manual intervals, pushing all the `request.send` promises into an array then doing `await Promise.all` could work.